common/iterator/FlatMappedIterator.java

```diff
@@ -73,5 +73,6 @@ public U next() {
     @Override
     public void recycle() {
         sourceIterator.recycle();
+        if (currentIterator != null) currentIterator.recycle();
```

Comment:
the active iterator we spawn from the `sourceIterator` must also be recycled, if there is one